upgrade.md

```diff
@@ -29,65 +25,56 @@ Provided you have suitable udev rules in play, this will already do everything y
 Download or compile a Black Magic Debug (BMD) firmware.
 As with the [bmputil](#bmputil) section above, the same considerations exist with this method for firmware selection.
 
+You will need to install [`dfu-util`](http://dfu-util.sourceforge.net/) package. On macOS you can use homebrew, macports or fink depending on your preferred package manager. You will need version 0.8.0 or greater to support the dfuse commands.
+
 Plug in the Black Magic Probe into your computer and run the following command:
 
 ```bash
 sudo dfu-util -d 1d50:6018,:6017 -s 0x08002000:leave -D blackmagic-native.bin
 ```
 
-You will need to install [`dfu-util`](http://dfu-util.sourceforge.net/) package. On MacOS you can use homebrew, macports or fink depending on your preferred package manager. You will need version 0.8.0 or greater to support the dfuse commands.
+To upgrade non-native hardware see the readme of the different [platforms on GitHub](https://github.com/blackmagic-debug/blackmagic/tree/main/src/platforms).
 
 ```{note}
 If `dfu-util` fails to switch your BMP into bootloader mode, or you feel like you might have **bricked** your BMP, you can also plug in your BMP while holding down the button. This will force the BMP to stay in the bootloader on power up.
 ```
 
-## Windows
+### Using stm32-mem.py
```

Comment:
This script is deprecated and going away in BMD v2.0 so should be removed from the instructions.

Reply:
It was removed.